### ISSUE TITLE
Suche nach String-Fragmenten in den Tatvorwürfen 

### ISSUE THREAD

**Issue by RalphBike** [weg-li/weg-li#755]
Die Suche nach den Tatvorwürfen ist bei der Fülle an Kandidaten nur semi-hilfreich. Bei den gefühlten 68 Einträgen findet man kaum den richtigen Kandidaten für "gehweg". Die Favoriten helfen auch nur begrenzt, da sie sich nicht an den Favoriten meines Users  orientieren und auch nicht steuerbar sind. 

Netter wäre eine Suche, die nicht nur den kompletten String, sondern und-verknüpft die einzelnen Fragmente des Strings sucht und gerne auch Quotes beherrscht. 

Z.B. 
```
gehweg behind => 102712, 112051, 112655, etc.
gehweg park stund => 112324, 112656, etc.
"es kam zum unfall" => Diverse Treffer
"es kam unfall" => Kein Treffer
```

Klingt kompliziert, ist mit Regex aber eigentlich ganz überschaubar und sogar Quotes lassen sich machen:
https://stackoverflow.com/questions/32034567/split-string-on-spaces-except-words-in-quotes

Wenn man ambitioniert genug ist, könnte man mit den Fragmenten dann sogar noch versuchen, ein paar Hilfestellungen zu geben, in dem für allgemein etablierte Begriffe auch das Behörden-Pendant gefunden wird.

z. B. 
```
zebra => Fußgängerüberweg
halteverbot => haltverbot
ampel => lichtzeigenanlage
behinderung => behindert
roller => elektrokleinstfahrzeug
scooter => elektrokleinstfahrzeug
gesenk => absenkung
gewicht => gesamtmasse
Radfahrstreifen => Zeichen 295
mittellinie => fahrstreifenbegrenzung
```

**Comment by phoet** [comment]
> Die Favoriten helfen auch nur begrenzt, da sie sich nicht an den Favoriten meines Users orientieren und auch nicht steuerbar sind.

Doch, genau das sind sie.

**Comment by phoet** [comment]
Die Community war ganz eindeutig für die Auswahl der tatsächlichen Vergehen. Das ist nun mal kompliziert.

**Comment by RalphBike** [comment]
Klar und an der Stelle könnte das Portal den User im Umgang mit den Tatvorwürfen besser unterstützen, dafür habe ich einen Vorschlag gemacht.

**Comment by phoet** [comment]
Die Daten des BKAT sind inherent schlecht strukturiert. Wenn die wenigstens eine Hierarchie hätten, dann könnte man schon wesentlich besser damit umgehen. Wir sind ja schon froh, dass wir die Daten über einen Umweg maschinell einlesen können.

**Comment by RalphBike** [comment]
Jo... ist bekannt. Die Bußgelder wirklich auch komplett willkürlich.

**Comment by phoet** [comment]
Nee, hab nur auf den falschen Knopf gedrückt
